lib/core/MongoDbUnresolvableTransactionStore.ts

```diff
@@ -1,67 +1,44 @@
-import { Collection, Db, Long, MongoClient } from 'mongodb';
 import IUnresolvableTransactionStore from './interfaces/IUnresolvableTransactionStore';
 import Logger from '../common/Logger';
-import MongoDbLogger from '../common/MongoDbLogger';
+import { Long } from 'mongodb';
+import MongoDbStore from '../common/MongoDbStore';
 import TransactionModel from '../common/models/TransactionModel';
 import UnresolvableTransactionModel from './models/UnresolvableTransactionModel';
 
 /**
  * Implementation of `IUnresolvableTransactionStore` that stores the transaction data in a MongoDB database.
  */
-export default class MongoDbUnresolvableTransactionStore implements IUnresolvableTransactionStore {
+export default class MongoDbUnresolvableTransactionStore extends MongoDbStore implements IUnresolvableTransactionStore {
   /** Collection name for unresolvable transactions. */
   public static readonly unresolvableTransactionCollectionName: string = 'unresolvable-transactions';
 
   private exponentialDelayFactorInMilliseconds = 60000;
   private maximumUnresolvableTransactionReturnCount = 100;
 
-  private db: Db | undefined;
-  private unresolvableTransactionCollection: Collection<any> | undefined;
-
   /**
-   * Constructs a `MongoDbUnresolvableTransactionStore`;
+   * Creates a new instance of this object.
+   * @param serverUrl The target server url.
+   * @param databaseName The database name where the collection should be saved.
    * @param retryExponentialDelayFactor
    *   The exponential delay factor in milliseconds for retries of unresolvable transactions.
    *   e.g. if it is set to 1 seconds, then the delays for retries will be 1 second, 2 seconds, 4 seconds... until the transaction can be resolved.
    */
-  constructor (private serverUrl: string, private databaseName: string, retryExponentialDelayFactor?: number) {
+  public constructor (
+    serverUrl: string,
+    databaseName: string,
+    retryExponentialDelayFactor?: number) {
```

Comment:
`retryExponentialDelayFactor` seems to be useless, remove it if it is bothering you.